File: cz/mg/vulkan/PFNvkCreateDescriptorPool.java
package cz.mg.vulkan;

public class PFNvkCreateDescriptorPool extends VkFunctionPointer {
    public PFNvkCreateDescriptorPool() {
    }

    public PFNvkCreateDescriptorPool(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkCreateDescriptorPool(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkCreateDescriptorPool(long value) {
        setValue(value);
    }

    public PFNvkCreateDescriptorPool(VkInstance instance) {
        super(instance, new VkString("vkCreateDescriptorPool"));
    }

    public void call(VkDevice device, VkDescriptorPoolCreateInfo pCreateInfo, VkAllocationCallbacks pAllocator, VkDescriptorPool pDescriptorPool, VkResult rval){
        call(getValue(), device != null ? device.getVkAddress() : VkPointer.getSinkAddress(), pCreateInfo != null ? pCreateInfo.getVkAddress() : VkPointer.NULL, pAllocator != null ? pAllocator.getVkAddress() : VkPointer.NULL, pDescriptorPool != null ? pDescriptorPool.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.getSinkAddress());
    }

    private static native void call(long vkaddress, long device, long pCreateInfo, long pAllocator, long pDescriptorPool, long rval);
}

File: 1.0/src/c/cz/mg/vulkan/PFN_vkCreateDescriptorPool.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkCreateDescriptorPool_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pCreateInfo, jlong pAllocator, jlong pDescriptorPool, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkCreateDescriptorPool f = (PFN_vkCreateDescriptorPool)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        ((VkDescriptorPoolCreateInfo*)jniLongToPointer(pCreateInfo)),
        ((VkAllocationCallbacks*)jniLongToPointer(pAllocator)),
        ((VkDescriptorPool*)jniLongToPointer(pDescriptorPool))
    );
}


jint Java_cz_mg_vulkan_PFNvkCreateDescriptorPool_callSimplifiedNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pCreateInfo, jlong pAllocator, jlong pDescriptorPool){
    (void)env;
    (void)jc;
    PFN_vkCreateDescriptorPool f = (PFN_vkCreateDescriptorPool)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        ((VkDescriptorPoolCreateInfo*)jniLongToPointer(pCreateInfo)),
        ((VkAllocationCallbacks*)jniLongToPointer(pAllocator)),
        ((VkDescriptorPool*)jniLongToPointer(pDescriptorPool))
    );
}

